src/database/size.ts

```diff
@@ -0,0 +1,115 @@
+import * as logger from "../logger";
+import * as pathLib from "path";
+
+import { ListRemote, RTDBListRemote } from "./listRemote";
+import { RTDBSizeRemote, SizeRemote } from "./sizeRemote";
+import { SizeResult } from "./sizeResult";
+import { Stack } from "../throttler/stack";
+
+/*
+ * For flat objects, performance suffers due to "slow-start"
+ * shallow gets. We set a generous initial list batch size
+ * to mitigate this.
+ */
+const LIST_BATCH_SIZE = 4000;
+
+/*
+ * Top-level GET requests for large subtrees will likely
+ * fail, so there are some probable gains to be had from
+ * performing shallow gets first. This constant controls
+ * how deep to search in the tree before attempting large
+ * GETs.
+ */
+const SKIP_DEPTH = 1;
+const TIMEOUT = 50;
+const TIMEOUT_STATUS_CODE = 400;
+
+/*
+ * Control how many outstanding full GETs and shallow
+ * GETs are allowed at a time. Wide trees will have
+ * substantially more "size" operations, whereas deep
+ * trees will have substantially more outstanding list
+ * operations.
+ */
+const SIZE_STACK_CONCURRENCY = 2000;
+const LIST_STACK_CONCURRENCY = 2000;
```

Comment:
A bit higher than I anticipate.

Remember RTDB only execute one action at a time.
Concurrent only help eat the network RTT.

I would do more experiment with deep-nested tree. I image this will the database.